File: experiments/llm-reasoning-cost/generated_code/go/05_file_io_mini.go
package main

import (
	"fmt"
	"os"
)

type Result struct {
	Message string
	Error   error
}

func writeGreeting(filename string) Result {
	file, err := os.Create(filename)
	if err != nil {
		return Result{"", err}
	}
	defer func() {
		if closeErr := file.Close(); closeErr != nil {
			err = closeErr
		}
	}()

	_, err = file.WriteString("Hello from the program!")
	if err != nil {
		return Result{"", err}
	}

	return Result{"File written successfully", nil}
}

func main() {
	result := writeGreeting("output.txt")
	if result.Error != nil {
		fmt.Println("Error:", result.Error)
	} else {
		fmt.Println(result.Message)
	}
}

File: experiments/llm-reasoning-cost/generated_code/rust/05_file_io_mini.rs
```rust
use std::fs::File;
use std::io::{self, Write};
use std::path::Path;

pub fn write_greeting(filename: &str) -> Result<(), io::Error> {
    let path = Path::new(filename);
    let mut file = File::create(&path)?;

    file.write_all(b"Hello from the program!")?;
    
    Ok(())
}
```